hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3248,6 +3257,86 @@ func isProgressing(ctx context.Context, hc *hyperv1.HostedCluster) (bool, error)
 	return false, nil
 }
 
+func (r *HostedClusterReconciler) validateAWSConfig(hc *hyperv1.HostedCluster) (string, error) {
+	if hc.Spec.Platform.Type != hyperv1.AWSPlatform {
+		return "", nil
+	}
+
+	if hc.Spec.Platform.AWS == nil {
+		return "", errors.New("aws cluster needs .spec.platform.aws to be filled")
+	}
+
+	var message string
+	// In a private topology all endpoints should be exposed as Routes.
+	// A common HCP router is exposed via Service type LB internal.
+	// Other publishing strategies are possible but not recommended.
+	if hc.Spec.Platform.AWS.EndpointAccess == hyperv1.Private {
+		for _, service := range []hyperv1.ServiceType{
+			hyperv1.APIServer,
+			hyperv1.Konnectivity,
+			hyperv1.OAuthServer,
+			hyperv1.OVNSbDb,
+			hyperv1.Ignition,
+			hyperv1.OIDC,
+		} {
+			service := hyperutil.ServicePublishingStrategyByTypeByHC(hc, service)
+			if service.Type != hyperv1.Route {
+				message = fmt.Sprintf("Unrecommended config for Private EndpointAccess: service %v with publishing strategy %v", service, service.Type)
```

Comment:
ok, I think there might scenarios where using other alternatives might come in handy e.g NodePort to reduce cloud resources.
However we can always relax the constraint later on. I'll update to fail if kas != lb/route and everything else != route.
Then I'll follow up later to default kas to route in private topology which we aren't doing yet, but I want to keep the PR scoped.

Reply:
Agreed that NodePort could be possible, but it is untested and undocumented how one would successfully do that.  Thus, I'm inclined to make that invalid so that no one say "I _can_ do this so I expect it to be supported".  We can always add intentional support/testing for it later.